fabrictestbed_extensions/fablib/interface.py

```diff
@@ -623,19 +638,6 @@ def get_vlan(self) -> str:
             vlan = None
         return vlan
 
-    def get_bandwidth(self) -> int:
-        """
-        Gets the FABRIC bandwidth of an interface.
-
-        :return: VLAN
-        :rtype: String
-        """
-        try:
-            bw = self.get_fim_interface().get_property(pname="capacities").bw
-        except:
-            bw = None
-        return bw
-
```

Comment:
Is this change intentional?  If so, since this is a public method, wouldn't it be better done in a separate change with some documentation (changelog, commit messages, etc) as to why this is being removed?

Reply:
There were multiple definitions for this function in this class. I removed one.

Reply:
Maybe there is a case for shadowing variables and methods in the same scope, but I wish Python would rather emit a warning in these cases...

Would have been useful to do this in a separate commit with a message, so that future readers won't have to puzzle over it. Oh well. :-)

Reply:
I know but for the sake of ease, going to leave it here :)

Reply:
I see only one definition of `Interface.get_bandwidth()`. I must be missing something?

https://github.com/fabric-testbed/fabrictestbed-extensions/blob/46c4397a464d4add245f633d4dca6e4150bdce40/fabrictestbed_extensions/fablib/interface.py#L547

Also it seems that ruff or flake8 should detect some cases of redefintions.  Perhaps we could target some level of lint cleanliness.

https://docs.astral.sh/ruff/rules/redefined-while-unused/

Reply:
May be that was fixed on main. All changes I have are based on rel1.7 which probably didn't have this fix.

Reply:
rel1.7 branch has two definitions one at line 600 and another at 626

Reply:
https://github.com/fabric-testbed/fabrictestbed-extensions/blob/6b2cc8dbeeadaa31d68bbc6bcb04c2d07771f004/fabrictestbed_extensions/fablib/interface.py#L600

https://github.com/fabric-testbed/fabrictestbed-extensions/blob/6b2cc8dbeeadaa31d68bbc6bcb04c2d07771f004/fabrictestbed_extensions/fablib/interface.py#L626

Reply:
Ah, thanks! I missed that this is being merged to rel1.7 branch.